src/core/activityContent/validation.ts

```diff
@@ -40,386 +42,388 @@ const SUPPORTED_IMAGE_MEDIA_TYPES = [
+  requireValidDimensions(obj, "ActivityContentImageLink");
+  return true;
+};
 
+const requireIsActivityContentVideoLink = (obj: unknown): obj is ActivityContentVideoLink => {
+  if (!isRecord(obj)) throw new InvalidActivityContentError(InvalidRecordMessage("ActivityContentVideoLink"));
+  requireIsMediaContentLink(obj, "ActivityContentVideoLink");
+  requireValidDimensions(obj, "ActivityContentVideoLink");
   return true;
 };
 
-/**
- * isActivityContentNote() is a type check function for ActivityContentNote
- * objects. Note that this function only checks that the given object meets the
- * type definition of an ActivityContentNote. It **does not** perform any logic
- * validations, such as checking the format of string fields or checking that
- * required attachments each include at least one supported format.
- *
- * @param obj - The object to check
- * @returns A boolean indicating whether or not the object is an ActivityContentNote
- */
-export const isActivityContentNote = (obj: unknown): obj is ActivityContentNote => {
-  if (!isRecord(obj)) return false;
-  if (obj["@context"] !== "https://www.w3.org/ns/activitystreams") return false;
-  if (obj["type"] !== "Note") return false;
-  if (!isString(obj["content"])) return false;
-  if (obj["mediaType"] !== "text/plain") return false;
-
-  if (obj["name"] && !isString(obj["name"])) return false;
-  if (obj["published"] && !isString(obj["published"])) return false;
-  if (obj["attachment"] && !isArrayOfType(obj["attachment"], isActivityContentAttachment)) return false;
-  if (obj["tag"] && !isArrayOfType(obj["tag"], isActivityContentTag)) return false;
-  if (obj["location"] && !isArrayOfType(obj["location"], isActivityContentLocation)) return false;
+const requireIsActivityContentLink = (obj: unknown): ActivityContentLink => {
+  if (!isRecord(obj)) throw new InvalidActivityContentError(InvalidRecordMessage("ActivityContentVideoLink"));
+  if (obj["type"] !== "Link") throw new InvalidActivityContentError("ActivityContentLink type is not 'Link'");
 
-  return true;
+  const href = requireIsString(obj["href"], "ActivityContentLink href");
+  if (!isValidHrefField(href)) throw new InvalidActivityContentError("ActivityContentLink href is invalid");
+
+  if (obj["name"]) requireIsString(obj["name"], "ActivityContentLink name");
+
+  return { type: "Link", href: "", ...obj };
 };
 
-/**
- * isActivityContentProfile() is a type check function for
- * ActivityContentProfile objects. Note that this function only checks that the
- * given object meets the type definition of an ActivityContentProfile. It
- * does not** perform any logic validations, such as checking the format of
- * string fields or checking that the icon field includes at least one supported
- * image format.
- *
- * @param obj - The object to check
- * @returns A boolean indicating whether or not the object is an ActivityContentProfile
- */
-export const isActivityContentProfile = (obj: unknown): obj is ActivityContentProfile => {
```

Comment:
Can we keep the non-error throwing versions of `isActivityContentNote` and `isActivityContentProfile` specifically? The have some value as type checks when we don't necessarily want to validate.

Reply:
Let's discuss what should be in those - I noticed some checks in there that seemed to conflate validation and type checking.  The error-throwing type checks for profile and note are quite a bit simpler now.  We could make non-error throwing versions of those maybe?

Reply:
yep, `isActivityContentNote` and `isActivityContentProfile` could just check that all the appropriate keys are present and have the correct type like they did before. they don't need to do regexes or anything.